hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -38,15 +44,24 @@ public class IngestChecker {
     private static final Logger LOG = LoggerFactory.getLogger(IngestChecker.class);
 
     private final AccountID nodeAccountID;
+    private final SignaturePreparer signaturePreparer;
+    private final Cryptography cryptography;
 
     /**
      * Constructor of the {@code IngestChecker}
      *
      * @param nodeAccountID the {@link AccountID} of the <em>node</em>
+     * @param signaturePreparer the {@link SignaturePreparer} that prepares signature data
+     * @param cryptography the {@link Cryptography} used to verify signatures
      * @throws NullPointerException if one of the arguments is {@code null}
      */
-    public IngestChecker(@NonNull final AccountID nodeAccountID) {
+    public IngestChecker(
+            @NonNull final AccountID nodeAccountID,
+            @NonNull final SignaturePreparer signaturePreparer,
+            @NonNull final Cryptography cryptography) {
         this.nodeAccountID = requireNonNull(nodeAccountID);
+        this.signaturePreparer = requireNonNull(signaturePreparer);
```

Comment:
It would be good to include clear messages to the requireNonNull method, so the thrown exception more clearly explains what went wrong, and where. (for example, "Null signature preparer provided to ingest checker in workflow; possible issue in workflow").

Reply:
I think the service team should introduce a helper method similar to what the platform team has. Until then, I think it is better to stick to the same pattern everywhere. This will make it easier to find the places that need to be updated.

Reply:
Objects.requireNonNull is already a helper method, why would we want to wrap that?  The description should be unique to the location where the check occurs...